src/video-player/components/runtime.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useRef, useEffect } from "react";
+import { IAuthoredState } from "./authoring";
+import { useRequiredQuestion } from "../../shared/hooks/use-required-question";
+import videojs from "video.js";
+import Shutterbug from "shutterbug";
+import css from "./runtime.scss";
+
+import "./video-js.css";
+
+export interface IInteractiveState {
+  percentageViewed: number;
+  lastViewedTimestamp: number;
+  submitted?: boolean;
+}
+
+interface IProps {
+  authoredState: IAuthoredState;
+  interactiveState?: IInteractiveState;
+  setInteractiveState?: (state: IInteractiveState) => void;
+  report?: boolean;
+  setNavigation?: (enableForwardNav: boolean, message: string) => void;
+}
+// small sample mp4
+// "https://models-resources.s3.amazonaws.com/geniblocks/resources/fablevision/video/charcoal.mp4";
+// sample captions
+// "https://models-resources.s3.amazonaws.com/question-interactives/test-captions.vtt";
+
+let saveStateInterval : any;
+
+export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState, setNavigation, report }) => {
+  let viewedProgress = interactiveState?.percentageViewed || 0;
+  let viewedTimestamp = interactiveState?.lastViewedTimestamp || 0;
+  const playerRef = useRef(null);
+  useEffect(() => {
+    loadPlayer();
+    Shutterbug.enable("." + css.runtime);
+    Shutterbug.on("saycheese", playerRef.current);
+    return () => {
+      Shutterbug.off("saycheese", playerRef.current);
+      Shutterbug.disable();
+    };
+  }, []);
+  const getViewTime = () => {
+    if (playerRef.current) {
+      const video: HTMLVideoElement = playerRef.current! as HTMLVideoElement;
+      return video.currentTime;
+    }
+    else return 0;
+  };
+  const getViewPercentage = () => {
+    if (playerRef.current) {
+      const video: HTMLVideoElement = playerRef.current! as HTMLVideoElement;
+      return video.currentTime / video.duration;
+    }
+    else return 0;
+  };
+
+  const { submitButton, lockedInfo } = useRequiredQuestion({ authoredState, interactiveState, setInteractiveState, setNavigation, isAnswered: viewedProgress > 0.96 });
```

Comment:
If you refactor this interactive to use `BaseQuestionApp`, this line won't be necessary.